src/main/java/net/snowflake/client/core/SFBaseSession.java

```diff
@@ -105,6 +105,17 @@
   // Stores other parameters sent by server
   private final Map<String, Object> otherParameters = new HashMap<>();
   private HttpClientSettingsKey ocspAndProxyKey = null;
+  // Memory limit for SnowflakeChunkDownloader. This gets set from SFBaseSession for testing
+  // purposes only.
+  private long memoryLimitForTesting = -1;
+
+  public void setMemoryLimitForTesting(long memLimit) {
```

Comment:
Added functions to manually change JDBC memory limit so that memory limit is low enough that chunk downloading prefetching has to be cancelled while still being able to run unit test in under 1 minute